### PR TITLE
BREAKING CHANGE: refactor: single wearable schema and multiple wearable types

### PR DESCRIPTION
**Pull request by guidota**
## Description 

Use single wearable schema for both third party or decentraland wearables. Two types also are provided `StandardWearable ` and `ThirdPartyWearable` containing required properties.

### Validations
Some validations were added to the `MerkleProof` object:
- 'hashingKeys' array can't be empty
- `proof` array can't be empty
- All `hashingKeys` must be present in the wearable metadata object
- Some properties are mutually exclusive, so it can be a Third Party or a regular Wearable but not both.

### CODE REVIEW

File: src/platform/wearables/wearable.ts [hunk 1]
```diff
@@ -4,33 +4,165 @@ import {
...
+        nullable: true
+      },
+      merkleProof: {
+        ...MerkleProof.schema,
+        nullable: true
+      },
+      content: {
+        type: 'object',
+        nullable: true
+      }
```
Comment (LautaroPetaccio) [inline src/platform/wearables/wearable.ts:146]: Content can be typed as a string -> string map:
```suggestion
      content: {
        type: 'object',
        additionalProperties: { type: 'string' },
        required: []
      },
```

File: src/platform/wearables/wearable.ts [hunk 2]
```diff
@@ -4,33 +4,165 @@ import {
...
+        type: 'string'
+      },
+      descriptions: {
+        type: 'array',
+        items: I18N.schema,
+        minItems: 1
+      },
       collectionAddress: {
+        type: 'string',
+        nullable: true
```
[the remark this answers is not in the document]
Comment (guidota) [reply src/platform/wearables/wearable.ts:90]: The reason is that 'ajv' complains with errors when optional properties are not marked as `nullable`.

Added some comments to the `validate` function to express these properties are mutually exclusive.

File: src/platform/wearables/wearable.ts [hunk 3]
```diff
@@ -4,33 +4,181 @@ import {
...
+  image: string
+  metrics?: Metrics
+  content?: any
+  merkleProof?: MerkleProof
 }
 
+/** @alpha */
+export type StandardWearable = Omit<
+  WithRequired<Wearable, 'collectionAddress' | 'rarity'>,
+  'merkleProof'
```
Comment (LautaroPetaccio) [inline src/platform/wearables/wearable.ts:42]: Standard wearables don't have the content property as well.

File: src/platform/wearables/wearable.ts [hunk 4]
```diff
@@ -4,33 +4,181 @@ import {
...
+    replaces: WearableCategory[]
+    hides: WearableCategory[]
+    tags: string[]
+    representations: WearableRepresentation[]
+    category: WearableCategory
+  }
+  thumbnail: string
+  image: string
+  metrics?: Metrics
+  content?: any
```
Comment (LautaroPetaccio) [inline src/platform/wearables/wearable.ts:35]:
```suggestion
  content?: Record<string, string>
```

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]

**Review by LautaroPetaccio — APPROVED** [review]

**Pull request merged**